core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -363,6 +363,15 @@ public boolean hasMandatoryDependencies() {
         return dependencies.stream().anyMatch(d -> !d.optional);
     }
 
+    /**
+     * Is this plugin labeled deprecated?
+     * @return {@code true} when plugin contains deprecated in categories, otherwise {@code false}.
+     * @since TODO
+     */
+    public boolean isDeprecated() {
+        return getInfo().isDeprecated();
```

Comment:
Potential null pointer, not sure it's the best approach calling `getInfo` to figure out if plugin is deprecated.
I would have to explore the call stack, However would be great if anyone has a better suggestion.